tests/components/freebox/test_device_tracker.py

```diff
@@ -0,0 +1,47 @@
+"""Tests for the Freebox sensors."""
+from unittest.mock import Mock
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.components.freebox import SCAN_INTERVAL
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.core import HomeAssistant
+
+from .common import setup_platform
+
+from tests.common import async_fire_time_changed
+
+
+async def test_router_mode(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+    router: Mock,
+) -> None:
+    """Test get_hosts_list invoqued multiple times if freebox into router mode."""
+    await setup_platform(hass, SENSOR_DOMAIN)
+
+    assert router().lan.get_hosts_list.call_count == 1
+
+    # Simulate an update
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    assert router().lan.get_hosts_list.call_count == 2
+
+
+async def test_bridge_mode(
+    hass: HomeAssistant,
+    freezer: FrozenDateTimeFactory,
+    router_bridge_mode: Mock,
+) -> None:
+    """Test get_hosts_list invoqued once if freebox into bridge mode."""
+    await setup_platform(hass, SENSOR_DOMAIN)
+
```

Comment:
```suggestion

    assert router_bridge_mode().lan.get_hosts_list.call_count == 1

```

Reply:
As the test above we test the same thing : initial + update state

So we understand the change between router and bridge mode